flask_hw3/blueprints/supermarkets/supermarket.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+from flask import Blueprint, render_template, request, session, url_for, redirect
+from werkzeug.utils import secure_filename
+
+from supermarket_db import SUPERMARKET_DB, SupermarketTemplate
+
+supermarket_page = Blueprint('supermarket_page', __name__,
+                             template_folder='./templates',
+                             static_folder='static',
+                             static_url_path='/supermarkets')
+
+
+@supermarket_page.route('/supermarket', methods=['GET', 'POST'])
+def get_or_update_supermarket():
+    if request.method == 'POST':
+        form = request.form
+        file = request.files['supermarket_img']
+        filename = secure_filename(file.filename)
+        if form:
+            product_to_add = SupermarketTemplate(img_name=filename, **form)
+            SUPERMARKET_DB.append(product_to_add)
+            file.save(os.path.join(supermarket_page.root_path + '/static/', filename))
```

Comment:
Do not concatenate strings using `+`. For python3 use `fstrings`, for python2 `.format()` method. It will be more readable and prevent some errors when you will add different data types

Reply:
fixed